backend/server/src/http.rs

```diff
@@ -43,21 +46,37 @@ pub(crate) async fn serve(
     // All our logic is encoded in the function `handle`. The only thing we are
     // doing here is to pass the context to that function, and clone its `Arc`
     // accordingly.
-    let factory = make_service_fn(move |_| {
-        let ctx = Arc::clone(&ctx);
-        async {
-            Ok::<_, Infallible>(service_fn(move |req| {
-                handle_panic(handle(req, Arc::clone(&ctx)))
-            }))
+    macro_rules! factory {
```

Comment:
```suggestion
    // 
    // We wrap the factory definition in a macro because we need two slightly 
    // different factories. One for binding to a unix socket and one for 
    // binding to a TCP socket. The code for defining the factory is exactly 
    // the same, but due to type inference, it results in a different type. The
    // macro avoids code duplication.
    macro_rules! factory {
```